journal_app/views.py

```diff
@@ -1,3 +1,167 @@
+from rest_framework.response import Response
+from rest_framework.authentication import TokenAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.status import HTTP_200_OK,HTTP_204_NO_CONTENT,HTTP_400_BAD_REQUEST,HTTP_404_NOT_FOUND,HTTP_201_CREATED
 # Create your views here.
+# on front end pick a food from foods and post to back end
+    # "foods": [
+    #     {
+    #         "fdcId": 2343477,
+    #         "description": "Pie, apple",
+    #         "commonNames": "",
+    #         "additionalDescriptions": "",
+    #         "dataType": "Survey (FNDDS)",
+    #         "foodCode": 53301000,
+    #         "publishedDate": "2022-10-28",
+    #         "foodCategory": "Cakes and pies",
+    #         "foodCategoryId": 2646625,
+    #         "allHighlightFields": "",
+    #         "score": 609.0303,
+    #         "microbes": [],
+    #         "foodNutrients": [
+    #             {
+    #                 "nutrientId": 1003,
+    #                 "nutrientName": "Protein",
+    #                 "nutrientNumber": "203",
+    #                 "unitName": "G",
+    #                 "value": 2.7,
+    #                 "rank": 600,
+    #                 "indentLevel": 1,
+    #                 "foodNutrientId": 28717129
+    #             },
+    #             {
+    #                 "nutrientId": 1004,
+    #                 "nutrientName": "Total lipid (fat)",
+    #                 "nutrientNumber": "204",
+    #                 "unitName": "G",
+    #                 "value": 15.3,
+    #                 "rank": 800,
+    #                 "indentLevel": 1,
+    #                 "foodNutrientId": 28717130
+    #             }
+class All_journals(APIView):
+    def get(self,request):
+        try:
+            all_journals = Journal_entry.objects.all()
+            serialized_journals = JournalEntrySerializer(all_journals, many=True)
+            serialized_journals_data = serialized_journals.data
+            for journal in serialized_journals_data:
+                plate_list = Plate.objects.filter(journal_entry_id__id__contains = journal['id'])
+                serialized_plates = PlateSerializer(plate_list,many=True)
+                journal['plates'] = serialized_plates.data
+                for plate_item in journal['plates']:
+                 ingredients = Ingredient.objects.filter(plate_id__id__contains = plate_item['id'])
+                 serialized_ingredients = IngredientSerializer(ingredients, many=True)
+                 plate_item['ingredients'] = serialized_ingredients.data
+                 for plate_ingredient in plate_item['ingredients']:
+                      ingredient_measurement = Measurement.objects.get(id=plate_ingredient['measurement_id'])
+                      serialized_measurement = MeasurementSerializer(ingredient_measurement)
+                      nutrients = Nutrient.objects.filter(id__in = plate_ingredient['nutrients_id'])
```

Comment:
Hey Marion I see you created the ability to have users. But none of the views in this file require token authentication. Wouldn't some of these action require the user to be authenticated?